<launch>
<include file="$(find rplidar_ros)/launch/rplidar_s3.launch"/>

<!-- static_transform_publisher x y z yaw pitch roll frame_id child_frame_id period_in_ms  rostopic echo -n5 /wall_angle  1.48 90   3.1416-->
<node pkg="tf" type="static_transform_publisher" name="base_to_laser"
      args="0.0 0.5 0  0 0 0  base_link laser 50"/> 




<node pkg="rosserial_python" type="serial_node.py" name="rosserial">
  <param name="port" value="/dev/ttyUSB1"/>
  <param name="baud" value="57600"/>
</node>


  <!-- 感知（RANSAC+质量指标） -->
  <node pkg="ransacbug2" type="perception.py" name="perception_lock_v2" output="screen">
    <param name="side" value="right"/>
    <param name="scan_topic" value="/scan"/>

  </node>

  <!-- 控制（ALIGN→LOCKED_GO→HOLD） -->
  <node pkg="ransacbug2" type="bug2_lock_v4.py" name="bug2_lock_v3" output="screen">
    <param name="side" value="right"/>
    <param name="state" value="ALIGN"/>
    <param name="scan_topic" value="/scan"/>
    <param name="cmd_vel_topic" value="/cmd_vel"/>
    <param name="perception_topic" value="/perception_lock_v2"/>
  </node>

</launch>